src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/OneVersusAllTrainer.cs

```diff
@@ -227,37 +228,324 @@ public override MulticlassPredictionTransformer<OneVersusAllModelParameters> Fit
+                var trainedData = new RoleMappedData(view, label: trainerLabel, feature: transformer.FeatureColumnName);
+
+                if (calibratedModel == null)
+                    calibratedModel = CalibratorUtils.GetCalibratedPredictor<TSubPredictor, TCalibrator>(Host, ch, Calibrator, (TSubPredictor)transformer.Model, trainedData, Args.MaxCalibrationExamples) as TDistPredictor;
+
+                Host.Check(calibratedModel != null, "Calibrated predictor does not implement the expected interface");
+                return new BinaryPredictionTransformer<TScalarPredictor>(Host, calibratedModel, trainedData.Data.Schema, transformer.FeatureColumnName);
+            }
+
+            return new BinaryPredictionTransformer<TScalarPredictor>(Host, transformer.Model, view.Schema, transformer.FeatureColumnName);
+        }
+        private protected override MulticlassPredictionTransformer<OneVersusAllModelParametersBase<CalibratedModelParametersBase<TSubPredictor, TCalibrator>>> FitHelper(IHost host, bool useProbabilities, TScalarPredictor[] predictors, DataViewSchema schema, string featureColumn, string labelColumnName)
+        {
+            return new MulticlassPredictionTransformer<OneVersusAllModelParametersBase<CalibratedModelParametersBase<TSubPredictor, TCalibrator>>>(Host, OneVersusAllModelParametersBuilder.Create(Host, useProbabilities, predictors.Cast<CalibratedModelParametersBase<TSubPredictor, TCalibrator>>().ToArray()), schema, featureColumn, LabelColumn.Name);
+        }
+
+        private protected override OneVersusAllModelParametersBase<CalibratedModelParametersBase<TSubPredictor, TCalibrator>> TrainCore(IChannel ch, RoleMappedData data, int count)
+        {
+            // Train one-vs-all models.
+            var predictors = new CalibratedModelParametersBase<TSubPredictor, TCalibrator>[count];
+            for (int i = 0; i < predictors.Length; i++)
+            {
+                ch.Info($"Training learner {i}");
+                predictors[i] = (CalibratedModelParametersBase<TSubPredictor, TCalibrator>)TrainOne(ch, Trainer, data, i).Model;
+            }
+            return OneVersusAllModelParametersBuilder.Create(Host, TrainerOptions.UseProbabilities, predictors);
+        }
+    }
+
+    /// <summary>
+    /// Strongly typed implementation of the <see cref="OneVersusAllTrainerBase{T}"/> where T is a <see cref="OneVersusAllModelParameters{T}"/>.  T can either be
+    /// a calibrated binary estimator of type <see cref="CalibratedModelParametersBase{TSubPredictor, TCalibrator}"/>, or a non calibrated binary estimary.
+    /// This cannot be used to turn a non calibrated binary classification estimator into its calibrated version. If that is required, use <see cref="OneVersusAllTrainerTyped{TSubPredictor, TCalibrator}"/> instead.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class OneVersusAllTrainerTyped<T> : OneVersusAllTrainerBase<OneVersusAllModelParametersBase<T>> where T : class
```

Comment:
I'm not sure having `Typed` at the end of this class name is the best. Why can't it just be `OneVersusAllTrainer<T>`?

Reply:
The problem is that the only difference between the original method and this one are the optional parameters. So if people are only specifying the required parameters, which is the most common case in our code base, then C# doesn't know which method to call. Isn't that a binary breaking change? So I chose to make this method have a different name. Do you have any other ideas on how to resolve it?

Reply:
Same question about the other method I added in. I called it `OneVersusAllUnCalibratedToCalibratedTyped`, but since it has 2 type parameters that have to be manually specified, its name can be anything and it wont have conflicts. I just named it this to be very clear on what it does, though I am happy to change it as well if you have thoughts on this too.

Reply:
This isn't a method, this is a class name. The class name and the method don't need to match 1-1.

If we need to suffix the method name with `Typed` or similar, that is fine. But the class doesn't need to have the suffix on it.

Reply:
I see. I was thinking of the Trainers Catalog when I responded. I have fixed the class names and will push those changes. Thanks!